### PR TITLE
Make port substitution for WebTransport over HTTP/3 server work

### PR DESCRIPTION
**Pull request by bashi**
Context: https://github.com/web-platform-tests/wpt/pull/29727#discussion_r700117465

This PR provides a default port for WebTransport over HTTP/3 server even when it is not enabled so that port substitutions work.

Another approach I tried is to handle missing ports in `template()` defined `tools/wptserve/wptserve/pipes.py` but it didn't work well since setting `request.server.config['ports']` correctly seems not trivial.

### CODE REVIEW

**Comment by bashi** [comment]
@jgraham Please take a look.

cc/ @yutakahirano @nidhijaju

**Review by foolip — APPROVED** [review]
Looks good to me, but I'll not merge since @jgraham's review was requested.

**Comment by bashi** [comment]
@jgraham friendly ping.

File: tools/serve/serve.py [hunk 1]
```diff
@@ -947,8 +951,9 @@ def build_config(logger, override_path=None, config_cls=ConfigBuilder, **kwargs)
     if enable_http2:
         rv._default["ports"]["h2"] = [9000]
 
-    if kwargs.get("webtransport_h3"):
-        rv._default["ports"]["webtransport-h3"] = [11000]
+    # Provide a default port number for WebTransport over HTTP/3 even it is not
```
Comment (foolip) [inline tools/serve/serve.py:954]:
```suggestion
    # Provide a default port number for WebTransport over HTTP/3 even if it is not
```

**Comment by foolip** [comment]
@jgraham I'll go ahead and merge this when CI passes, but still feel free to review before or after then.

File: tools/serve/serve.py [hunk 2]
```diff
@@ -947,8 +951,9 @@ def build_config(logger, override_path=None, config_cls=ConfigBuilder, **kwargs)
     if enable_http2:
         rv._default["ports"]["h2"] = [9000]
 
-    if kwargs.get("webtransport_h3"):
-        rv._default["ports"]["webtransport-h3"] = [11000]
+    # Provide a default port number for WebTransport over HTTP/3 even if it is not
+    # enabled so that port substitutions work.
+    rv._default["ports"]["webtransport-h3"] = [11000]
```
Comment (jgraham) [inline tools/serve/serve.py:956]: Look at `ConfigBuilder` around like 861, I think we can just put the default as "auto" in that.

**Review by jgraham — CHANGES_REQUESTED** [review]

File: tools/serve/serve.py (hunk 2 above)
Comment (bashi) [reply tools/serve/serve.py]: Thank you! I wasn't aware "auto" selects an available` port automatically by opening a socket. That's much better. Done.

**Review by jgraham — APPROVED** [review]

**Comment by foolip** [comment]
I'll go ahead and merge, the "Azure Pipelines (infrastructure/ tests: macOS)" failures are unrelated, and also aren't required checks.

**Comment by bashi** [comment]
Thank you for review jgraham, and thank you for merging foolip!

**Pull request merged**